libs/ec/inc/public/tfc/devices/base.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <boost/asio.hpp>
+
+#include "tfc/ecx.hpp"
+
+namespace tfc::ec::devices {
+class base {
+public:
+  explicit base(boost::asio::io_context& ctx) : ctx_(ctx) {}
+  static constexpr auto vendor_id = 0x0;
```

Comment:
what type is deduced as auto here?

Reply:
These are both unsigned 32-bit integers. Changed this to a concrete type for all devices.